lib/ZnapZend/Config.pm

```diff
@@ -156,18 +157,38 @@ my $checkBackupSets = sub {
 my $getBackupSet = sub {
     my $self = shift;
     my $enabledOnly = shift;
-    my $dataSet = shift;
-    
-    #get all backup sets and check if valid
-    $self->backupSets($self->zfs->getDataSetProperties($dataSet));
+    # The recursion setting allows to find datasets under the named one
+    # (e.g. a pool root DS that might not necessarily have a znapzend
+    # configuration by itself). Similar to listing ALL configs when no
+    # dataset was passed, but no impact of looking at the whole system.
+    my $recurse = shift;
+
+    #get all backup sets and check if valid, from remainder of ARGV
+    if (scalar(@_) > 0) {
+        $self->backupSets($self->zfs->getDataSetProperties(\@_, $recurse));
+    } else {
+        # Not that recursion makes much sense for "undef" (=> list everything)
+        $self->backupSets($self->zfs->getDataSetProperties(undef, $recurse));
+    }
     $self->$checkBackupSets();
 
+    printf STDERR "=== getBackupSet() : got " . scalar(@{$self->backupSets}) . " dataset(s) with a local backup plan\n" if $self->debug;
+    # Note/FIXME? If there were ZFS errors getting some of several
+    # requested datasets, but at least one succeeded, the result is OK.
+    if (scalar(@{$self->backupSets}) == 0) {
+        return 0; # false
+    }
+
     if ($enabledOnly){
         my @backupSets;
 
         for my $backupSet (@{$self->backupSets}){
             push @backupSets, $backupSet if $backupSet->{enabled} eq 'on';
         }
+        printf STDERR "=== getBackupSet() : got " . scalar(\@backupSets) . " enabled-only dataset(s) with a local backup plan\n" if $self->debug;
+        if (scalar(\@backupSets) == 0) {
```

Comment:
```suggestion
        if (not @backupSets == 0) {
```

Reply:
\@backupSets is a pointer to an array ... scalar of a pointer to an array would always be 1

Reply:
Curiously, the code did work, bot for the `print` above (returning 0 or 1 or 5 entries as expected) and for the conditional here.

Reply:
Oh, so maybe I did get the context wrong - this is the lower check, for enabledOnly stuff.

Reply:
Posted https://github.com/oetiker/znapzend/pull/439/commits/8b20412a36edffd154feb9306c00a7ed197af8fa